### DISCUSSION TITLE
get_domain_area

### DISCUSSION THREAD

**Discussion by kizzle4shaw** [PMEAL/OpenPNM#2723] [Q&A]
I extracted a network using snow2. I'm trying to calculate permeability, when I use the get_domain_area function it gives me a warning and error, how do I get an accurate area measurement?

![image](https://user-images.githubusercontent.com/32852010/229582407-7880f3e9-b21c-4db6-bd07-4db204a23014.png)

**Comment by jgostick** [comment]
I'm sorry, but the dev team is *very small*.  It consists of myself, whomever is doing their PhD with me (on PNMs), and occasionally a postdoctoral associate. In reality that means 2-3 people are working on OpenPNM and PoreSpy at any given time, and it is therefore not feasible for us to provide tech support for most questions.  We have to focus our attention on fixing bugs and moving forward with our own projects.  Hopefully one of the other users will step in to answer your question, or you're able to answer it using your own resources. [^1]

[^1]: This long response was 'auto-generated' using the [saved replies](https://docs.github.com/en/get-started/writing-on-github/working-with-saved-replies/using-saved-replies) feature.

**Reply by gxw-ll** [reply]
Hi Prof, when using the code A = op.topotools.get_domain_area(pn, inlets=pn.pores('xmin'), outlets=pn.pores('xmax')), I encountered a similar issue. In my pore network model, I found that pn['pore.coords'][pn.pores('xmin')] is a one-dimensional array with a size of 81, while pn['pore.coords'][pn.pores('xmax')] is a one-dimensional array with a size of 84. Is this due to not correctly capturing the pores?

**Reply by jgostick** [reply]
funnily enough, my above comment still applies.  This problem is due to your lack of understanding python/openpnm, sorry I can't do a tutorial for you.

**Reply by gxw-ll** [reply]
Thank you for your prompt response, Prof, I just checked the source code. The function get_domain_area uses the ConvexHull function from the scipy library to calculate the area. Specifically, hull_in = ConvexHull(points=inlets[:, Nin]) calculates the area of the inlet face, and hull_out = ConvexHull(points=outlets[:, Nout]) calculates the area of the outlet face. Due to the irregularity of my pore network model, the areas of the inlet and outlet faces are not equal. It seems that we choose to use the area based on the inlet as a reference. Is that correct?

**Reply by jgostick** [reply]
The get_domain_area is only useful if you don't already know it.  In your case you know it from the size/shape of the image from which the network was extracted.

**Reply by gxw-ll** [reply]
Great! Thank you for your response.
